File: typescript_discord_bot_startup_sl8s/app/src/model/example/list_example.ts
import { BaseListModel } from "@antonpichka/vitruvius";
import Example from "./example";

class ListExample<T extends Example> extends BaseListModel<T> {
    public constructor(listModel: Array<T>) {
        super(listModel);
    }

    public override clone(): ListExample<T> {
        const newListModel = new Array<T>();
        for(const itemModel of this.listModel) {
            newListModel.push(itemModel.clone() as T);
        }
        return new ListExample(newListModel);
    }

    public override toString(): string {
        let strListModel = "\n";
        for(const itemModel of this.listModel) {
            strListModel += itemModel + ",\n";
        }
        return "ListExample(listModel: [" + strListModel + "])";
    }
}

export default ListExample;

File: dart_web_startup_sl8s/app/lib/model/example/list_example.dart
import 'package:dart_web_startup_sl8s/model/example/example.dart';
import 'package:library_architecture_mvvm_modify/library_architecture_mvvm_modify.dart';
import 'package:meta/meta.dart';

@immutable
base class ListExample<T extends Example> extends BaseListModel<T> {
  const ListExample({required List<T> listModel}) : super(listModel);

  @override
  ListExample<T> clone() {
    List<T> newListModel = List.empty(growable: true);
    for (final T itemModel in listModel) {
      newListModel.add(itemModel.clone() as T);
    }
    return ListExample<T>(listModel: newListModel);
  }

  @override
  String toString() {
    String strListModel = "\n";
    for (final T itemModel in listModel) {
      strListModel += "$itemModel,\n";
    }
    return "ListExample(listModel: [$strListModel])";
  }
}